bson/src/main/scala/types.scala

```diff
@@ -492,15 +442,12 @@ object BSONObjectID {
     id(3) = (timestamp & 0xFF).toByte
 
     if (!fillOnlyTimestamp) {
-      // machine id, 3 first bytes of md5(macadress or hostname)
-      id(4) = machineId(0)
-      id(5) = machineId(1)
-      id(6) = machineId(2)
-
-      // 2 bytes of the pid or thread id. Thread id in our case. Low endian
-      val threadId = Thread.currentThread.getId.toInt
-      id(7) = (threadId & 0xFF).toByte
-      id(8) = (threadId >> 8 & 0xFF).toByte
+      id(4) = (randomVal1 >> 16 & 0xFF).toByte
```

Comment:
Technically you don't even need the `& 0xFF` part, since it happens automatically.